src/main/java/edu/cornell/kfs/sys/dataaccess/impl/DocumentMaintenanceDaoJdbc.java

```diff
@@ -0,0 +1,131 @@
+package edu.cornell.kfs.sys.dataaccess.impl;
+
+import edu.cornell.kfs.sys.CUKFSParameterKeyConstants;
+import edu.cornell.kfs.sys.batch.DocumentRequeueStep;
+import edu.cornell.kfs.sys.dataaccess.DocumentMaintenanceDao;
+import org.kuali.kfs.coreservice.framework.parameter.ParameterService;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.rice.core.framework.persistence.jdbc.dao.PlatformAwareDaoBaseJdbc;
+import org.kuali.rice.kew.actionrequest.ActionRequestValue;
+import org.kuali.rice.kew.api.KEWPropertyConstants;
+import org.kuali.rice.kew.api.KewApiConstants;
+import org.kuali.rice.kew.routeheader.DocumentRouteHeaderValue;
+import org.springframework.util.StringUtils;
+
+import javax.persistence.Column;
+import javax.persistence.Table;
+import java.lang.annotation.Annotation;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+public class DocumentMaintenanceDaoJdbc extends PlatformAwareDaoBaseJdbc implements DocumentMaintenanceDao {
+	private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(DocumentMaintenanceDaoJdbc.class);
+
+	public static final String WORKFLOW_DOCUMENT_HEADER_ID_SEARCH_RESULT_KEY = "documentId";
+
+    private ParameterService parameterService;
+
+	/**
+	 * @see DocumentMaintenanceDao#getDocumentRequeueValues()#getDocumentRequeueFileValues()
+	 */
+	@Override
+	public Collection<String> getDocumentRequeueValues() {
+		String requeueSqlQuery = buildRequeueSqlQuery();
+		List<Map<String, Object>> results = getSimpleJdbcTemplate().queryForList(requeueSqlQuery);
+
+		return convertResultsMapListToListOfDocumentIds(results);
+	}
+
+	private String buildRequeueSqlQuery() {
+		StringBuilder sql = new StringBuilder();
+
+		sql.append("select DOC_HDR_ID from CYNERGY.KREW_DOC_HDR_T where ");
+		sql.append(retrieveColumnNameFromAnnotations(DocumentRouteHeaderValue.class, "docRouteStatus"));
+		sql.append("='");
+		sql.append(KewApiConstants.ROUTE_HEADER_ENROUTE_CD);
+		sql.append("' AND ");
+		sql.append(retrieveColumnNameFromAnnotations(DocumentRouteHeaderValue.class, KEWPropertyConstants.DOCUMENT_TYPE_ID));
+		sql.append(" NOT IN (");
+
+		final Collection<String> docTypeIds = parameterService.getParameterValuesAsString(DocumentRequeueStep.class, CUKFSParameterKeyConstants.NON_REQUEUABLE_DOCUMENT_TYPES);
+		sql.append(StringUtils.collectionToCommaDelimitedString(docTypeIds));

```

Comment:
These two lines are appending parameter-configured values in an unsafe manner; there's the potential for SQL injection if someone manages to update the parameter accordingly.